src/main/java/com/aliyun/odps/jdbc/OdpsResultSet.java

```diff
@@ -195,6 +195,18 @@ public Object getObject(int columnIndex) throws SQLException {
 
     Object obj = row[columnIndex-1];
     wasNull = (obj == null);
+
+    // odps-sdk views STRING as byte[], but JDBC can only see String
+    if (obj instanceof byte[]) {
+      try {
+        String charset = stmt.getConnection().getCharset();
+        if (charset != null) {
+          return new String((byte[]) obj, charset);
```

Comment:
This implementation could break `OdpsResultSet.getBytes()`. Maybe we should extract the original implementation as a private `getInnerObject`, do the cast String thing in `getObject`, and make sure `getBytes` works as expected.